src/js/actions/tools.js

```diff
@@ -418,7 +418,7 @@ define(function (require, exports) {
     selectTool.reads = [];
     selectTool.writes = [locks.JS_TOOL, locks.PS_TOOL];
     selectTool.transfers = [resetBorderPolicies, policy.removePointerPolicies, installShapeDefaults,
-        policy.removeKeyboardPolicies, policy.addPointerPolicies, policy.addKeyboardPolicies,
+        policy.removeKeyboardPolicies, policy.addPointerPolicies, policy.addKeyboardPolicies, policy.setMode,
```

Comment:
yes, it is required; I triggered an error of not doing so and then realized that `selectTool.transfers` is actually a superset of actions, including those used in each tool's `selectHandler` and `deselectHandler`. For example, `policy.setMode`, `policy.addKeyboardPolicies`, `shortcuts.addShortcut`